strawberryfields/engine.py

```diff
@@ -336,8 +357,159 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
+    """
+
+    HARDWARE_BACKENDS = ("chip0",)
+
+    def __init__(self, polling_delay_seconds=1, **kwargs):
+        # Only chip0 backend supported initially.
+        backend = "chip0"
+        super().__init__(backend)
+
+        api_client_params = {k: v for k, v in kwargs.items() if k in APIClient.DEFAULT_CONFIG}
+        self.client = APIClient(**api_client_params)
+        self.polling_delay_seconds = polling_delay_seconds
+        self.jobs = []
+
+    def __str__(self):
+        return self.__class__.__name__ + "({})".format(self.backend_name)
+
+    def _init_backend(self, *args):
+        """
+        TODO: This does not do anything right now.
+        """
+        # Do nothing for now...
+        pass
+
+    def reset(self):
+        """
+        Reset must be called in order to submit a new job. This clears the job queue as well as
+        any ran Programs.
+        """
+        super().reset(backend_options={})
+        self.jobs.clear()
+
+    def _get_blackbird(self, name, shots, program):
+        """
+        Returns a Blackbird object to be sent later to the server when creating a job.
+        Assumes the current backend as the target.
+
+        Args:
+            name (str): the name of the job to be created (e.g. StateTeleportation)
+            shots (int): the number of shots
+            blackbird_code: the blackbird code of the job
```

Comment:
```suggestion
            program (Program): program to be converted to Blackbird code
```

Reply:
Thanks, this was stale, left over from the old implementation.